packages/database/src/migrations/20210126013657-AddK13C580YPositiveMapOverlay-modifies-data.js

```diff
@@ -0,0 +1,139 @@
+const NEW_OVERLAY_GROUP_ID = generateId();
+const NEW_OVERLAY_GROUP_RELATION_FOR_OVERLAY_ID = generateId();
+const NEW_OVERLAY_GROUP_RELATION_FOR_GROUP_ID = generateId();
+
+const BASE_OVERLAY = {
+  userGroup: 'STRIVE User',
+  dataElementCode: 'value',
+  isDataRegional: true,
+  measureBuilderConfig: {
+    dataValues: {
+      STR_K13C05: 1,
+    },
+    programCode: 'SK13C',
+    entityAggregation: {
+      aggregationType: 'REPLACE_ORG_UNIT_WITH_ORG_GROUP',
+      dataSourceEntityType: 'village',
+      aggregationEntityType: 'facility',
+    },
+  },
+  measureBuilder: 'countEventsPerOrgUnit',
+  countryCodes: '{PG}',
+  projectCodes: '{strive,explore}',
+};
+
+const LINKED_MEASURE_OVERLAY = {
+  ...BASE_OVERLAY,
+  id: `${LINKED_MEASURE_OVERLAY_ID}`,
+  name: 'K13 C580Y Positive Count',
+  presentationOptions: {
+    values: [
+      {
+        value: 'other',
+        hideFromLegend: true,
+        color: 'green',
+      },
+      {
+        value: '0',
+        hideFromLegend: true,
+        color: 'red',
+      },
+      {
+        value: null,
+        hideFromLegend: true,
+        color: 'grey',
+      },
+    ],
+    displayType: 'color',
+    measureLevel: 'Facility',
+  },
+};
+
+const MAIN_OVERLAY = {
+  ...BASE_OVERLAY,
+  id: `${MAIN_OVERLAY_ID}`,
+  name: 'K13 C580Y Positive',
+  linkedMeasures: `{${LINKED_MEASURE_OVERLAY_ID}}`,
+  presentationOptions: {
+    values: [
+      {
+        name: 'Positive result',
+        color: 'green',
+        value: 'other',
+        hideFromPopup: true,
+      },
+      {
+        name: 'No positive result',
+        color: 'red',
+        value: '0',
+        hideFromPopup: true,
+      },
+      {
+        name: 'No data',
+        color: 'grey',
+        value: null,
+        hideFromPopup: true,
+      },
+    ],
+    displayType: 'color',
+    measureLevel: 'Facility',
+  },
+};
+
+const NEW_OVERLAY_GROUP = {
+  id: `${NEW_OVERLAY_GROUP_ID}`,
+  name: 'STRIVE Molecular Data',
+  code: 'STRIVE_Molecular_Data',
+};
+
+const NEW_OVERLAY_GROUP_RELATION_FOR_OVERLAY = {
+  id: `${NEW_OVERLAY_GROUP_RELATION_FOR_OVERLAY_ID}`,
+  map_overlay_group_id: `${NEW_OVERLAY_GROUP_ID}`,
+  child_id: `${MAIN_OVERLAY_ID}`,
+  child_type: 'mapOverlay',
+  sort_order: '0',
+};
+
+const NEW_OVERLAY_GROUP_RELATION_FOR_GROUP = {
+  id: `${NEW_OVERLAY_GROUP_RELATION_FOR_GROUP_ID}`,
+  map_overlay_group_id: '5f88d3a361f76a2d3f000004',
+  child_id: `${NEW_OVERLAY_GROUP_ID}`,
+  child_type: 'mapOverlayGroup',
+};
+
+exports.up = async function(db) {
+  await insertObject(db, 'mapOverlay', LINKED_MEASURE_OVERLAY);
+  await insertObject(db, 'mapOverlay', MAIN_OVERLAY);
+  await insertObject(db, 'map_overlay_group', NEW_OVERLAY_GROUP);
+  await insertObject(db, 'map_overlay_group_relation', NEW_OVERLAY_GROUP_RELATION_FOR_OVERLAY);
+  await insertObject(db, 'map_overlay_group_relation', NEW_OVERLAY_GROUP_RELATION_FOR_GROUP);
+};
+
+exports.down = function(db) {
```

Comment:
I definitely reckon you could write an `exports.down` for this. I think it should be as simple as deleting all the records you just inserted right?
(https://beyondessential.slack.com/archives/CDGS6P56K/p1590559208005500)

Reply:
I thought so too, however I was having issues with the `map_overlay_group_relation` and `map_overlay_group` deletions yesterday. I assumed it was foreign key constraints but I looked into it more this morning and it seems the issue is because I create their `id` using the `generateId()` function

For example, I set the `id` using `const NEW_OVERLAY_GROUP_ID = generateId();`

When I do it this way neither `DELETE FROM "map_overlay_group" WHERE "id" = '${NEW_OVERLAY_GROUP.id}'` or `DELETE FROM "map_overlay_group" WHERE "id" = '${NEW_OVERLAY_GROUP_ID}'` will work (the code will run, however these entries will not be deleted). However, if I change the `id` to:

`const NEW_OVERLAY_GROUP_ID = 'test1234'`

then it will work.

To solve this, I just changed the problematic DELETEs to not refer to the `id` & instead refer to something else unique - I've just pushed this code to this PR, let me know if it looks okay

Reply:
Good point! I think it is somewhat related to my first comment (that running code at file level can sometimes be confusing).